src/utilities.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Return true if given addresses are the same
+ * @param address1 the first address to test
+ * @param address2 the second address to test
+ */
+export function addressesAreEquals(address1: string, address2: string) {
+  return (
+    address1 && address2 && address1.toLowerCase() === address2.toLowerCase()
+  );
+}
+
+/**
+ * Search given address in given block's transactions (`from` field or `to` field).
+ * Return matching transactions hash into an array.
+ * @param {string} address
+ * @param {any} block
+ * @returns {string[]} an array containing all matching transactions hash
+ */
+export function findAddressInBlock(address: string, block: any): string[] {
```

Comment:
This method is actionable, so we should let the user decides when to execute it (in order to delay execution for eg.).
We also could have a much more composable design of functions.

Current function signature:
```typescript
(string, any) => string[]
```

What do we want as a user of trackr-lib? To be able:
- choose filtering algorithm from input address (`address` parameter)
- to provide next work to do on each block found (thanks to `newTxCallback` parameter)

Proposal:

`trackr-lib.ts`
```typescript
export function watch(
  nodeUrl: string,
  address: string,
  newTxCallback: (tx: string) => void,
  errorCallback: (error: Error) => void
): void {
  const web3 = new Web3(nodeUrl).eth;

  // This filter check '.to' and '.from' address from 'transaction'
  let filterAddressesAreEquals = Utilities.filterAllAddressesIntoTransaction(address);

  let findAddressInBlocks = Utilities.findAddressInBlock(filterAddressesAreEquals, newTxCallback);

  web3
    .subscribe("newBlockHeaders")
    .on("data", (blockHeader: any) => {
      web3.getBlock(blockHeader.number, true).then(findAddressInBlocks)
    })
    .on("error", errorCallback);
}
```





`utilities.ts`
```typescript
/**
 * Return true if given addresses are the same
 * @param address1 the first address to test
 * @param address2 the second address to test
 */
function addressesAreEquals(address1: string, address2: string): boolean {
  let result = false;
  if (address1 && address2) {
    result = address1.toLowerCase() === address2.toLowerCase()
  }
  return result;
}

export function filterAddressesAreEqualsTo(address1: string): (address: string) => boolean {
  return (address2: string) => {
    return addressesAreEquals(address1, address2);
  }
}

export function filterAllAddressesIntoTransaction(address: string): (transaction: any) => boolean {
  return (transaction: any) => {
    let filterAddress = filterAddressesAreEqualsTo(address);
    return filterAddress(transaction.from) ||
      filterAddress(transaction.to);
  }
}

export function findAddressInBlock(filter: (transaction: any) => boolean, newTxCallback: (tx: string) => void): ((block: any) => void) {
  return (block: any) => {
    return block.transactions
      .filter(filter)
      .map((transaction: any) => transaction.hash)
      .forEach((hash: string) => newTxCallback(hash));
  }
}
```